documentation/course/e2/README.md

```diff
@@ -631,38 +671,69 @@ TODO @tchojnacki
     <th>Wady</th>
   </tr>
   <tr>
-    <th>Rozwiązanie 1</th>
+    <th>Komunikacja API Gateway → Account Service przy każdym zapytaniu</th>
     <td>
       <ul>
-        <li>Zaleta 1</li>
+        <li>Zachowanie SRP</li>
+        <li>Brak dociążenia API Gateway</li>
+        <li>Brak potrzeby dzielenia kluczy publicznych</li>
```

Comment:
```suggestion
        <li>Brak potrzeby dzielenia kluczy publicznych</li>
        <li>Możliwość zastosowania prostego <i>reverse-proxy</i></li>
```

Reply:
Nope, the API Gateway would have to be an app either way (with solutions numbered as in the table):
- **Solution 1:** the gateway would have to make a call to Account Service and then make a decision based on the response, outside of nginx's scope
- **Solution 2:** the gateway would have to decode JWTs based on public keys fetched and cached from Account Service, outside of nginx's scope
- **Solution 3:** the gateway would have to make a call to the sidecar and then make a decision based on the response, outside of nginx's scope (note how this is exactly the same as Solution 1, except you make a call to the sidecar instead of main service, the sidecar doesn't complicate gateway's logic in any way)

The inability to use reverse-proxy is a downside, but it's a downside of all three solutions, not one in particular. We didn't lose the way to introduce reverse-proxy with the introduction of sidecar, but rather when we decided that there is no internal authorization between services, so when we decided to check all permissions in API Gateway instead of checking them in each service separately - this was decided during the presentation of your diagram to the supervisor. And note that this old solution is neither of presented in the table.

Reply:
Yup, my bad.